### ISSUE TITLE
FR: -H option like grep

### ISSUE THREAD

**Issue by kristopolous** [draios/sysdig#682]
So in grep you can do -H in order to show what file something is in.  Let me show you the use-case I'm dealing with.

I have a long running sysdig with the options `-C 20 -W 10` and occasionally when the hiesenbug I'm dealing with occurs I then log in and scp them off the machine to some large directory where I'm looking for patterns.

In practice what I end up doing is something like

```
ls | xargs -n 1 sysdig fd.type=file and evt.failed=true -r
```
over, in this case, 112 files.  Sure enough things are found.  

There's two issues (feel free to spin off bugs)
1. I can easily script this and write headers to find out where the file is from ... but it would be nice to have something like grep's -H option

2. For some reason I can't just cat the files on to each other and use that as a giant capture file. I know I know, I wrote this file cycling feature to begin with ... but I'm curious as to whether this is just me being careless in using what I wrote, or if this genuinely doesn't work (it really should).

really, you can just assign (2) to me if you feel like it.

**Comment by kristopolous** [comment]
Ok so for (2) these are not truncated files.

```
$ sysdig fd.type=file and evt.failed=true -r cd24c5de925bc8e24eecf6636df29811
$ sysdig fd.type=file and evt.failed=true -r 1cd0d196ca97f24f6db66e0025d9e35a
$ cat cd24c5de925bc8e24eecf6636df29811 1cd0d196ca97f24f6db66e0025d9e35a > bigger
$ sysdig fd.type=file and evt.failed=true -r bigger
res = 1
unexpected block type 168627466
$
```

Maybe I don't know the file type that well, but if this is anything like the ntar/pcap days from yesteryear, something like the operation above should be ok.

**Comment by ldegio** [comment]
regarding (2), merging files is not as trivial as catting them, because each file contains metadata blocks (process list, interface list...) that need to be parsed before ingesting the events.

In the ideal world, the metadata parsing could happen at any point in the capture, but in reality at this time a lot of assumptions are made that this happens only once when the capture start.

It's possible modify the code so that metadata blocks that show up after the initialization are skipped. This would sort of work (especially in your scenario), but would likely cause the loss of some state.
